### PR TITLE
fix(fiori-elements-writer): manifest separation ui5 libs

### PR DESCRIPTION
**Pull request by martinjanus-sap**
#1942

### CODE REVIEW

File: packages/ui5-application-writer/src/types.ts [hunk 1]
```diff
@@ -63,6 +63,7 @@ export interface UI5 {
     typesPackage: string;
     descriptorVersion: string;
     ui5Libs: string | string[];
+    manifestLibs?: string | string[];
```
Comment (IainSAP) [inline packages/ui5-application-writer/src/types.ts:66]: @martinjanus-sap Can you please document the changes and the behaviour for consumers of the API please? Adding tsdocs here should suffice but please also the `ui5Libs` property.

Comment (martinjanus-sap) [reply packages/ui5-application-writer/src/types.ts:100]: Please check commit - https://github.com/SAP/open-ux-tools/pull/2280/commits/d65f0bd376fa569dd30b78456abcacbb882d6664 

For some types I tried my best to infer usage and give appropriate doc

**Review by longieirl — DISMISSED** [review]
Good work.
Test coverage is good.
Additional documentation is added.

**Review by longieirl — DISMISSED** [review]
re-approving.

**Review by IainSAP — DISMISSED** [review]
Thx @martinjanus-sap 

- Updates are clear, new type property is well doc'd
- Tests updated as expected

File: packages/fiori-freestyle-writer/test/__snapshots__/basic.test.ts.snap [hunk 2]
```diff
@@ -3287,7 +3286,7 @@ export function createDeviceModel () {
     <!-- Bootstrap the UI5 core library. 'data-sap-ui-frameOptions=\\"allow\\"'' is a NON-SECURE setting for test environments -->
     <script id=\\"sap-ui-bootstrap\\"
         src=\\"../resources/sap-ui-core.js\\"
-        data-sap-ui-libs=\\"sap.m,sap.ui.core,sap.ushell\\"
+        data-sap-ui-libs=\\"sap.m,sap.ui.core\\"
```
Comment (devinea) [inline packages/fiori-freestyle-writer/test/__snapshots__/basic.test.ts.snap:3289]: Need to check into this. It doesn't look right to me. flpSandbox.html should always have `sap.ushell`

Comment (martinjanus-sap) [reply packages/fiori-freestyle-writer/test/__snapshots__/basic.test.ts.snap]: Handled with suggested code - https://github.com/SAP/open-ux-tools/pull/2280/commits/97603f60d27230aef1529859db018a9441a332b2

Thanks

File: packages/fiori-freestyle-writer/src/defaults.ts [hunk 3]
```diff
@@ -45,6 +45,17 @@ export function setDefaults(ffApp: FreestyleApp<unknown>): void {
     }
     // All fiori-freestyle apps should use load reuse libs, unless explicitly overridden
     ffApp.appOptions = Object.assign({ loadReuseLibs: true }, ffApp.appOptions);
+    if (ffApp.ui5) {
```
Comment (IainSAP) [inline packages/fiori-freestyle-writer/src/defaults.ts:48]: Why is this necessary here but not in fiori-elements-writer?

Comment (devinea) [reply packages/fiori-freestyle-writer/src/defaults.ts:48]: But there is something similar in FE https://github.com/SAP/open-ux-tools/pull/2280/files#diff-e26bb8b0304f6b67d4b5d0479d6c75263405883302396f777dd7f1ac6cb37211R35

**Review by devinea — APPROVED** [review]
Thanks @martinjanus-sap

File: packages/fiori-freestyle-writer/src/defaults.ts (hunk 3 above)
Comment (IainSAP) [reply packages/fiori-freestyle-writer/src/defaults.ts:48]: Same outcome but I think a clearer implementation. We can take that as a separate alignment task.

**Review by IainSAP — APPROVED** [review]
Re-approving
